types/d3-sankey/index.d.ts

```diff
@@ -0,0 +1,267 @@
+// Type definitions for D3JS d3-sankey module 0.4
+// Project: https://github.com/d3/d3-sankey/
+// Definitions by: Tom Wanzek <https://github.com/tomwanzek>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+// Last module patch version validated against: 0.4.2
+
+/**
+ * A helper interface as an extension reference for user-provided properties of
+ * nodes and links in the graph, which are not required or calculated by
+ * the Sankey Layout Generator
+ */
+export interface SankeyExtraProperties { [key: string]: any; }
```

Comment:
Very good question. I wasn't to happy with it either. Here is the rationale:

It prevents silly things like:  `SankeyNode<number, string>`, i.e. using primitives instead of something that has a key-value structure.

So you're right, dropping it won't fail any tests or meaningful use, it only prevents an edge case failure mode.

(_Side note_: Sooner or later we might want to consider opening up a discussion issue regarding creating and publishing a TS2.2+ conforming definition, e.g. here we could use default assignments of `{}` for the generics. I just dread 😰  the parallel maintenance period of the definitions, even if we can shorten some signatures, consider using `keyof` etc.)